stdlib/public/core/Hasher.swift

```diff
@@ -251,6 +251,16 @@ extension Hasher {
   }
 }
 
+#if $Embedded
+@usableFromInline
+var _swift_stdlib_Hashing_parameters: _SwiftHashingParameters = {
```

Comment:
The idea is that the user / the toolchain maintainer handles that in arc4random_buf.

Reply:
note: one consequence is that if you declare any reference type you then MUST implement arc4_random because you then support classes which require Hashable transitively

Reply:
Not accurate -- see <https://github.com/apple/swift/pull/70344/files#diff-1289fc8d34bfbbd931d86529683eb2b8f54c9c5d3d6b3f6111fa192d6650e71a> where a test shows that even if you declare classes/subclasses, that alone doesn't trigger dependency on Hashable / arc4_random. Only if you actually ask for a hash of something (directly, or by using a Set) do you get the dependency.

Reply:
I think Kuba is right here -- hashing is not a baseline thing for native Swift classes. (It is part of the NSObject contract, but native Swift classes do not need to opt into that.)

Reply:
Turning this into a Swift variable is somewhat risky -- global variables in Swift are subject to concurrency-related constraints that may be overkill for this use case.

The seed is used every time we hash anything, so it ought to be trivially accessible, preferably with absolutely no synchronization overhead (or any branches, even).

Reply:
The constraints you're talking about are essentially a swift_once wrapper, correct? And the concern with that is about performance, but not correctness, right?

If so, I'm inclined to use this mechanism for its simplicity, and leave the performance concern as a future/followup improvement. Unconditionally initializing the hash seed won't be great for embedded Swift users who don't use hashing.